### DISCUSSION TITLE
Custom Payment Gateway Currency not supported

### DISCUSSION THREAD

**Discussion by hakimj96** [impress-org/givewp#6315] [Developer Q&A]
Hello!

I've created a custom GiveWP payment gateway plugin and tested out completely, it was flawless in our website. but in merchant website, the error comes out as "MYR" is not supported for our payment gateway. Additional note, during testing I was using the same currency as the merchant tested. So, any ideas where should I began checking? I notice I did not declare any currency supported in our plugin. So I am really lost right now. Plus, I the merchant had enable multiple currencies (premium add on) option. Our website is using basic GiveWP since we only install for testing only. Does the multiple currencies do effect the whole process?

The repo is [here](https://github.com/technicalbpm/givewp-berrypay).

**Comment by kjohnson** [comment]
@hakimj96

My first question would be if the gateway supports "MYR" - that would need to be confirmed.

Let's start without Currency Switcher to narrow down the issue, since it sounds like your test site was only using GiveWP (the main plugin).

Do you have a sandbox site that we could look at? I'm not able to test a custom gateway without gateway credentials.
